src/shogun/base/SGObject.cpp

```diff
@@ -993,6 +1000,16 @@ bool CSGObject::equals(const CSGObject* other) const
 	{
 		const BaseTag& tag = it.first;
 		const Any& own = it.second.get_value();
+
+		if (!own.visitable())
+		{
+			SG_SDEBUG(
+			    "Skipping comparison of %s::%s of type %s as it is "
+			    "non-visitable.\n",
+			    this->get_name(), tag.name().c_str(), own.type().c_str());
+			continue;
```

Comment:
same here, i would move this down and only say "skipping" to avoid duplicate code